crates/types/src/data.rs

```diff
@@ -446,25 +445,20 @@ pub struct Leaf<TYPES: NodeType> {
 }
 
 impl<TYPES: NodeType> Leaf<TYPES> {
+    #[allow(clippy::unused_async)]
     /// Calculate the leaf commitment,
     /// which is gated on the version to include the block header.
     pub async fn commit<V: Versions>(
         &self,
-        upgrade_lock: &UpgradeLock<TYPES, V>,
+        _upgrade_lock: &UpgradeLock<TYPES, V>,
     ) -> Commitment<Self> {
-        let version = upgrade_lock.version_infallible(self.view_number).await;
-
-        if version < V::Marketplace::VERSION {
-            <Self as Committable>::commit(self)
-        } else {
-            RawCommitmentBuilder::new("leaf commitment")
-                .u64_field("view number", *self.view_number)
-                .field("parent leaf commitment", self.parent_commitment)
-                .field("block header", self.block_header.commit())
-                .field("justify qc", self.justify_qc.commit())
-                .optional("upgrade certificate", &self.upgrade_certificate)
-                .finalize()
-        }
+        RawCommitmentBuilder::new("leaf commitment")
```

Comment:
Can we just `<Self as Committable>::commit(self)` here to avoid duplication?

Reply:
done!